app/src/main/java/com/sdp/movemeet/utility/ImageHandler.java

```diff
@@ -1,87 +1,118 @@
 package com.sdp.movemeet.utility;
 
+import android.app.Activity;
+import android.graphics.Bitmap;
+import android.graphics.drawable.Drawable;
 import android.net.Uri;
 import android.util.Log;
 import android.view.View;
+import android.widget.ImageView;
 import android.widget.ProgressBar;
 
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 
-import com.google.android.gms.tasks.OnFailureListener;
-import com.google.android.gms.tasks.OnSuccessListener;
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.request.target.CustomTarget;
+import com.bumptech.glide.request.transition.Transition;
 import com.google.android.gms.tasks.Task;
 import com.google.firebase.storage.UploadTask;
 import com.sdp.movemeet.backend.BackendManager;
 import com.sdp.movemeet.backend.firebase.storage.StorageImageManager;
 import com.sdp.movemeet.models.Image;
-import com.squareup.picasso.Picasso;
+import com.sdp.movemeet.view.activity.ActivityDescriptionActivity;
+import com.sdp.movemeet.view.activity.ActivityDescriptionActivityUnregister;
+import com.sdp.movemeet.view.profile.EditProfileActivity;
+import com.sdp.movemeet.view.profile.ProfileActivity;
+import static com.sdp.movemeet.utility.ActivityPictureCache.loadFromCache;
+import static com.sdp.movemeet.utility.ActivityPictureCache.saveToCache;
+import static com.sdp.movemeet.utility.PermissionChecker.isStorageReadPermissionGranted;
+import static com.sdp.movemeet.utility.PermissionChecker.isStorageWritePermissionGranted;
 
 /**
  * This class allows to both load images from and upload images to the Firebase Storage service.
  * In case the image is not present in the local cache, it is fetched from Firebase Storage.
  * On the contrary, if the the image is already in the local cache, it is simply loaded from there.
  */
-public class ImageHandler {
+public abstract class ImageHandler {
 
-    private static final String TAG = "FirebaseInteraction";
+    private static final String TAG = "ImageHandler";
 
     private static BackendManager<Image> imageBackendManager;
 
+
     /**
      * Fetch an image (user profile picture, activity header picture or chat image) from Firebase Storage.
      *
      * @param image Image object to be loaded from Firebase Storage or from the local cache.
-     * @param progressBar A ProgressBar displayed as long as the image is not yet fetched.
+     * @param activity Activity on which we want to fetch the image
```

Comment:
Here we can see the change you brought to the signature of the `loadImage` and `uploadImage` methods of the `ImageHandler` class 👍